### DISCUSSION TITLE
Delete image in Photos app

### DISCUSSION THREAD

**Discussion by rdeoliveira** [RhetTbull/osxphotos#1725] [Q&A]
Is it possible to delete an image in Photos (eg with its UUID)?

**Comment by RhetTbull** [comment]
No this isn't currently possible. You could add them to an album then delete later:

`osxphotos query --uuid UUID --add-to-album "delete me"`

**Reply by rdeoliveira** [reply]
Thanks! 

I'm looking to automate a process with Python script, so is there a Python API to achieve the above? 

For now I've been successful using `subprocess`:

```python
subprocess.run([
  "osxphotos",
  "query",
  "--uuid", 
  image.uuid,
  "--add-to-album", 
  "\"delete me\""
])
```

but wondering if there's a more elegant way? Also, I quite like to test scripts from my IDE (PyCharm) as I code, but the OS (Sequoia) doesn't allow me to fiddle with Photos from another (3rd party) app; it works from terminal.

**Reply by oPromessa** [reply]
@rdeoliveira Check also --uuid-from-file option should you have a list of uuids to "tag for deletion". 

For automations via python check examples/cli_example2.py

(Edit)
> my IDE (PyCharm) as I code, but the OS (Sequoia) doesn't allow me to fiddle with Photos from another (3rd party) app

I run it on visual studio code IDE terminal window on Sequoia without issue.

**Reply by rdeoliveira** [reply]
@oPromessa thanks! Great shout about `--uuid-from-file`; added to backlog. I had seen `cli_example2.py`, indeed useful. Yeah frustrating that PyCharm doesn't have the same permissions as VSC it seems; if I find out how to grant it the necessary permission, I will post the solution here.

**Reply by RhetTbull** [reply]
> my IDE (PyCharm) as I code, but the OS (Sequoia) doesn't allow me to fiddle with Photos from another (3rd party) app

I don't use PyCharm but check `Settings > Privacy & Security > Photos`  and see if PyCharm is listed (and disabled). If not, then it is likely PyCharm doesn't include the right permissions in their app's `Info.plist`, which is an oversight on their part. An editor for macOS should include all possible permissions and entitlements your code might need because it's the editor that is responsible for requesting permission from macOS when running your code in the editor.

**Reply by rdeoliveira** [reply]
@RhetTbull good idea; PyCharm wasn't listed and there is no `+` button (or similar) to add more apps to the list; Terminal is in the list (which explains why I can run my script from Terminal). It's OK though; it may slow debugging, but I can work with PyCharm + Terminal. Thanks!

**Reply by RhetTbull** [reply]
It might be worth opening an issue with the PyCharm developers.
